gcloud-java-storage/src/test/java/com/google/gcloud/storage/it/ITStorageTest.java

```diff
@@ -307,10 +308,18 @@ public void testListBlobsSelectedFields() {
     Page<Blob> page = storage.list(BUCKET,
         Storage.BlobListOption.prefix("test-list-blobs-selected-fields-blob"),
         Storage.BlobListOption.fields(BlobField.METADATA));
-    int index = 0;
+    // Listing blobs is eventually consistent, we loop until the list is of the expected size. The
+    // test fails if timeout is reached.
+    while (Iterators.size(page.values().iterator()) != 2) {
```

Comment:
I used `values().iterator()` simply because I expected list of 2 elements. But you are right we better use `iterateAll` that covers all cases.